larreco/RecoAlg/TrackMomentumCalculator.cxx

```diff
@@ -38,37 +38,97 @@ using std::endl;
-    {10,    14,    20,    30,    40,     80,     100,    140,    200,   300,
-     400,   800,   1000,  1400,  2000,   3000,   4000,   8000,   10000, 14000,
-     20000, 30000, 40000, 80000, 100000, 140000, 200000, 300000, 400000}};
-  TGraph const KEvsR{29, Range_grampercm.data(), KE_MeV.data()};
+  constexpr std::array<double, 73> KE_MeV{
+    {10.0,    12.0,    14.0,    17.0,    20.0,    25.0,    30.0,    35.0,    40.0,    45.0,
+     50.0,    55.0,    60.0,    70.0,    80.0,    90.0,    100.0,   120.0,   140.0,   170.0,
+     200.0,   250.0,   300.0,   350.0,   400.0,   450.0,   500.0,   550.0,   600.0,   700.0,
+     800.0,   900.0,   1000.0,  1200.0,  1400.0,  1700.0,  2000.0,  2500.0,  3000.0,  3500.0,
+     4000.0,  4500.0,  5000.0,  5500.0,  6000.0,  7000.0,  8000.0,  9000.0,  10000.0, 12000.0,
+     14000.0, 17000.0, 20000.0, 25000.0, 30000.0, 35000.0, 40000.0, 45000.0, 50000.0, 55000.0,
+     60000.0, 70000.0, 80000.0, 90000.0, 1e+05,   1.2e+05, 1.4e+05, 1.7e+05, 2e+05,   2.5e+05,
+     3e+05,   3.5e+05, 4e+05}};
+  TGraph const KEvsR{73, Range_grampercm.data(), KE_MeV.data()};
   TSpline3 const KEvsR_spline3{"KEvsRS", &KEvsR};
 
+  // Stopping power data from pdg, to be used with MCS
+  const std::vector<double> dedx_GeV_per_cm()
+  {
+    // original table (in MeV cm2/g)
+    std::vector<double> dEdx{
+      {5.687, 4.979, 4.461, 3.902, 3.502, 3.042, 2.731, 2.508, 2.34,  2.21,  2.107, 2.023, 1.954,
+       1.848, 1.771, 1.713, 1.67,  1.609, 1.57,  1.536, 1.519, 1.508, 1.51,  1.517, 1.526, 1.537,
+       1.548, 1.559, 1.57,  1.591, 1.61,  1.628, 1.645, 1.675, 1.7,   1.733, 1.761, 1.799, 1.829,
+       1.855, 1.877, 1.897, 1.914, 1.93,  1.944, 1.969, 1.991, 2.01,  2.028, 2.058, 2.084, 2.119,
+       2.149, 2.193, 2.232, 2.269, 2.304, 2.337, 2.369, 2.4,   2.43,  2.49,  2.548, 2.606, 2.663,
+       2.776, 2.888, 3.055, 3.224, 3.498, 3.774, 4.052, 4.332}};
+    for (double& value : dEdx) {
+      value *= LAr_density * 1e-3; // convert to GeV/cm
+    }
+    return dEdx;
+  }
+
+  const int ndedx = 73;
+  const std::vector<double> dEdx_GeV_per_cm = dedx_GeV_per_cm();
+  const std::vector<double> E_GeV{
+    {0.115695, 0.117697, 0.119693, 0.122694, 0.125695, 0.13069,  0.135694, 0.14069,  0.145714,
+     0.150689, 0.155682, 0.160666, 0.165693, 0.17566,  0.185714, 0.1957,   0.205644, 0.225683,
+     0.245698, 0.275669, 0.305658, 0.355669, 0.405711, 0.45563,  0.505671, 0.555646, 0.605694,
+     0.655676, 0.705661, 0.805664, 0.905689, 1.00557,  1.10606,  1.30529,  1.50571,  1.8061,
+     2.10565,  2.60614,  3.1058,   3.60555,  4.10536,  4.60521,  5.10609,  5.606,    6.10591,
+     7.10579,  8.10569,  9.10561,  10.1106,  12.1105,  14.1104,  17.1103,  20.1103,  25.1102,
+     30.1102,  35.1102,  40.1101,  45.1101,  50.1101,  55.1101,  60.1101,  70.1101,  80.1101,
+     90.1101,  100.1,    120.1,    140.1,    170.1,    200.1,    250.1,    300.1,    350.1,
+     400.1}};
+  TGraph const dEdx_vs_E{ndedx, &E_GeV[0], &dEdx_GeV_per_cm[0]};
```

Comment:
Suggest replacing `&E_GeV[0]` with `E_GeV.data()` (and the same for `dEdx_GeV_per_cm`).